src/app/features/profile/store/profile.effects.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { loadProfile } from '@store/actions';
+import { map, mergeMap } from 'rxjs/operators';
+import { UserProfile } from '@interfaces';
+import { loadProfileSuccess } from '@store/actions';
+import { ProfileService } from '@core/services/profile.service';
+
+@Injectable() export class ProfileEffects {
+
+    loadProfile$ = createEffect(() => {
+
+        return this.actions$.pipe(
+                ofType(loadProfile),
+                mergeMap((action) => {
+
+                    return this.profileService.getProfile().pipe(
+                        map(( data: any ) => {
+
+                            for (const key in data) {
+
+                                const profileState: UserProfile = data[key].user;
+
+                                const profile = { user: profileState };
+                                return loadProfileSuccess( profile );
```

Comment:
this is a strange way to load users
usually when you need a state for the page it is called users or userProfiles not a single user
and then you are making 10 requests and submit an action with all the users not an action per each user